packages/paste-core/components/disclosure/__tests__/disclosure.test.tsx

```diff
@@ -184,12 +186,18 @@ describe('Disclosure', () => {
       expect(renderedDisclosureButton.getAttribute('aria-disabled')).toEqual('true');
       expect(renderedDisclosureButton.getAttribute('tabindex')).toEqual('0');
     });
-    it('should render a disclosure open and update attributes when clicked using a state hook', () => {
+    it('should render a disclosure open and update attributes when clicked using a state hook', async () => {
       render(<StateHookMock />);
-      const renderedDisclosureButton = screen.getByRole('button');
-      expect(renderedDisclosureButton.getAttribute('aria-expanded')).toEqual('true');
-      fireEvent.click(renderedDisclosureButton);
-      expect(renderedDisclosureButton.getAttribute('aria-expanded')).toEqual('false');
+      const toggleButton = screen.getByTestId('external-toggle');
+      const disclosureButton = screen.getByTestId('disclosure-button');
+      const disclosureContent = screen.getByTestId('disclosure-content');
+      expect(disclosureButton.getAttribute('aria-expanded')).toEqual('false');
+      expect(disclosureContent).not.toBeVisible();
+      await waitFor(() => {
+        userEvent.click(toggleButton);
+      });
+      expect(disclosureButton.getAttribute('aria-expanded')).toEqual('true');
+      expect(disclosureContent).toBeVisible();
```

Comment:
This is a much better test for the state hook. The previous one just mimicked the default behaviour, it tested nothing.

This now at least, tests that you can use the state hook to affect the behaviour of the disclosure by controlling is externally using one of the returned hook methods.

Reply:
🪙 🪙 [@testing-library/react-hooks](https://github.com/testing-library/react-hooks-testing-library) is super useful for unit testing hooks and make things simpler/less opaque.